lib/consumer.js

```diff
@@ -64,15 +54,8 @@ Consumer.prototype.getRegistrar = function() {
  */
 Consumer.prototype.getRegistrant = function(address) {
   var self = this;
-  return new Promise(function(fulfill, reject) {
-    self.registrar.registrantIndex.call(address, {from: self.address}, function(err, index) {
-      if (err) {
-        console.error(err);
-        reject(err);
-        return;
-      }
-      fulfill(self.getRegistrantbyIndex(index));
-    });
+  return self.registrar.registrantIndexAsync(address, {from: self.address}).then(function(index) { 
```

Comment:
You can make this even shorter by getting rid of the `self` variable. I also like to have my `then blocks start on a new line and you can use the arrow functions to remove boilerplate. So something like
```
return this.registrar.registrantIndexAsync(address, {from: this.addresss})
.then(index => this.getRegistrantbyIndex(index));
```